src/OpenTK.Windowing/NativeWindow.cs

```diff
@@ -62,60 +57,241 @@ public bool Visible
+                return new Size(width, height);
+            }
+            set => throw new NotImplementedException();
+        }
 
-        public WindowState WindowState { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public WindowBorder WindowBorder { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public Rectangle Bounds { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public Point Location { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public Size Size { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public int X { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public int Y { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public int Width { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public int Height { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public Rectangle ClientRectangle { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public Size ClientSize { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public MouseCursor Cursor { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public bool CursorVisible { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public bool CursorGrabbed { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public MouseCursor Cursor
+        {
+            get
+            {
+                throw new NotImplementedException();
 
+            }
+            set => throw new NotImplementedException();
+        }
 
-        public NativeWindow(int width = 640, int height = 480, string title = "", bool windowedFullscreen = false)
+        private bool _cursorHidden;
+        public bool CursorVisible
+        {
+            get
+            {
+                unsafe
+                {
+                    return glfw.GetInputMode(windowPtr, InputMode.GLFW_CURSOR) == InputModeValue.GLFW_CURSOR_HIDDEN
+                           || glfw.GetInputMode(windowPtr, InputMode.GLFW_CURSOR) ==
+                           InputModeValue.GLFW_CURSOR_DISABLED;
+                }
+            }
+            set
+            {
+                _cursorHidden = value;
+                unsafe
+                {
+                    if (value)
+                    {
+                        glfw.SetInputMode(windowPtr, InputMode.GLFW_CURSOR, InputModeValue.GLFW_CURSOR_HIDDEN);
```

Comment:
I think my understanding problem lies here then:
Because if we set CursorVisible to true, I would expect it to set the InputModeValue to GLFW_CURSOR_NORMAL and not ..._HIDDEN. Otherwise the property should be called CursorHidden/CursorInvisible, should it not?(Same goes for the getter)...

Reply:
You're totally right, I have that backwards. This also brings into question the behavior if it's never initialized. We would have to check the visible status on init and set accordingly.
The backing field still seems necessary; let me know what you think.

Reply:
Fixed by latest commit